### PR TITLE
Feat test components

### PR DESCRIPTION
**Pull request by jiixing**
initial unit tests for meditation.component and message.component

### CODE REVIEW

File: src/app/commitment/testing/fake-commitment.service.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import {TestHelper} from '../../../testing/test.helper'
+export class FakeCommitmentService {
+
+  public constructor() {
```
Comment (binarious) [inline src/app/commitment/testing/fake-commitment.service.ts]: Please remove the empty constructor from every class.

File: src/app/commitment/testing/fake-commitment.service.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  public getAll() {
+    return TestHelper.noRespose()
+
```
Comment (binarious) [inline src/app/commitment/testing/fake-commitment.service.ts]: Please remove the empty line.

File: src/app/commitment/testing/fake-commitment.service.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        "minutes": 20,
+        "users": ["3", "4", "5"],
+        "__v": 224
+      }
+    )
+  }
+
+  public save(commitment) {
+    return TestHelper.noRespose()
+
```
Comment (binarious) [inline src/app/commitment/testing/fake-commitment.service.ts]: Please remove the empty line.

File: src/app/commitment/testing/fake-commitment.service.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import {TestHelper} from '../../../testing/test.helper'
```
Comment (binarious) [inline src/app/commitment/testing/fake-commitment.service.ts]: There needs to be spacing after `{` and before `}`. Can you please fix this in your files?
Example:
```ts
import { TestHelper } from ...
```

File: src/app/meditation/meditation.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import {FakeUserService} from '../user/testing/fake-user.service';
+import {CommitmentService} from '../commitment/commitment.service';
+import {FakeCommitmentService} from '../commitment/testing/fake-commitment.service';
+import {FormsModule} from '@angular/forms';
+import {MockComponent} from '../../testing/mock-component';
+import {AvatarDirective} from '../profile/avatar.directive';
+import {AppState} from '../app.service'
+import {MeditationComponent} from './meditation.component';
+import {addMatchers} from '../../testing/jasmine-matchers'
+describe('MeditationComponent', () => {
```
Comment (binarious) [inline src/app/meditation/meditation.component.spec.ts]: Please add an empty line before this.

File: src/app/meditation/meditation.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  let fixture: ComponentFixture<MeditationComponent>;
+
+  beforeEach(async(() => {
+    addMatchers()
+    TestBed.configureTestingModule({
+
+      imports: [MaterialModule,
+        FormsModule,
+        NoopAnimationsModule,
+
```
Comment (binarious) [inline src/app/meditation/meditation.component.spec.ts]: Please remove the empty line.

File: src/app/meditation/meditation.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+import {addMatchers} from '../../testing/jasmine-matchers'
+describe('MeditationComponent', () => {
+  let component: MeditationComponent;
+  let fixture: ComponentFixture<MeditationComponent>;
+
+  beforeEach(async(() => {
+    addMatchers()
+    TestBed.configureTestingModule({
+
+      imports: [MaterialModule,
```
Comment (binarious) [inline src/app/meditation/meditation.component.spec.ts]: `MaterialModule` needs to go to the next line.

File: src/app/meditation/meditation.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+import {MeditationComponent} from './meditation.component';
+import {addMatchers} from '../../testing/jasmine-matchers'
+describe('MeditationComponent', () => {
+  let component: MeditationComponent;
+  let fixture: ComponentFixture<MeditationComponent>;
+
+  beforeEach(async(() => {
+    addMatchers()
+    TestBed.configureTestingModule({
+
```
Comment (binarious) [inline src/app/meditation/meditation.component.spec.ts]: Please remove the empty line.

File: src/app/meditation/meditation.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+      imports: [MaterialModule,
+        FormsModule,
+        NoopAnimationsModule,
+
+        RouterTestingModule
+      ],
+
+      declarations: [MeditationComponent, AvatarDirective,
```
Comment (binarious) [inline src/app/meditation/meditation.component.spec.ts]: Please just one declaration per line.

File: src/app/meditation/meditation.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MeditationComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should be created', () => {
+    expect(component).toBeTruthy();
+    const compiled = fixture.debugElement.nativeElement;
+
```
Comment (binarious) [inline src/app/meditation/meditation.component.spec.ts]: Please remove the empty lines.

File: src/app/meditation/meditation.component.spec.ts [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+  });
+
+  it('should have disabled submit button',()=>{
+    const compiled = fixture.debugElement.nativeElement;
+    expect(
+      compiled.querySelector('button[type="submit"]').disabled
+    ).toBe(true);
+
```
Comment (binarious) [inline src/app/meditation/meditation.component.spec.ts]: Please remove the empty line.

File: src/app/meditation/meditation.component.spec.ts [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+  })
+  it('should have 20 minute commitment',()=>{
+    const compiled = fixture.debugElement.nativeElement;
+    let commitment =compiled.querySelector('.commitment')
+    debugger
+    expect(commitment).toHaveText('20 minutes daily')
+
```
Comment (binarious) [inline src/app/meditation/meditation.component.spec.ts]: Please remove the empty line.

File: src/app/meditation/testing/fake-meditation.service.ts [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+      "walkingLeft": 0,
+      "sittingLeft": 0,
+      "status": "done"
+    }])
+  }
+
+  public getTimes() {
+    return TestHelper.fakeResponse(
+
+      {"0":2448,"1":4340,"2":2179,"3":3186,"4":3999,"5":3502,"6":5516,"7":5810,"8":3806,"9":2936,"10":4659,"11":4966,"12":3758,"13":3720,"14":4125,"15":6568,"16":9166,"17":4655,"18":2703,"19":2202,"20":4874,"21":4021,"22":3718,"23":2464}
```
Comment (binarious) [inline src/app/meditation/testing/fake-meditation.service.ts]: This line's too long. Please break it into several lines.

File: src/app/message/message.component.spec.ts [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+      ],
+      declarations: [MessageComponent,
+        MockComponent({selector: 'emoji-select'}),
+        MockComponent({selector: 'message-list-entry', inputs: ['message', 'admin', 'menuOpen']}),
+      ],
+      providers: [
+        {provide: MessageService, useClass: FakeMessageService},
+        {provide: WebsocketService, useClass: FakeWebsocketService},
+        {provide: UserService, useClass: FakeUserService}]
+
```
Comment (binarious) [inline src/app/message/message.component.spec.ts]: Please remove the empty line.

File: src/app/message/message.component.spec.ts [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+import {MaterialModule} from '@angular/material';
+
+describe('MessageComponent', () => {
+  let component: MessageComponent;
+  let fixture: ComponentFixture<MessageComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [MaterialModule,
```
Comment (binarious) [inline src/app/message/message.component.spec.ts]: New line

File: src/app/shared/testing/fake-websocket.service.ts [hunk 16]
```diff
@@ -0,0 +1,18 @@
+import {TestHelper} from '../../../testing/test.helper'
+import { Observable } from 'rxjs/Rx';
+
+export class FakeWebsocketService {
+
```
Comment (binarious) [inline src/app/shared/testing/fake-websocket.service.ts]: Please remove the empty lines.

**Comment by binarious** [comment]
Thanks a lot. Could you please run `ng lint` locally? I think there will be some errors.

File: README.md [hunk 17]
```diff
@@ -23,4 +23,11 @@ export let ApiConfig = {
 ```
 yarn
 ng serve
-```
\ No newline at end of file
+```
+
+## Testing Development Tips
```
Comment (binarious) [inline README.md]: Suggestion:
<pre><code>To isolate a spec file per run, change the function name `it` to `fit` or `describe` to `fdescribe`.
To speed up compilation, turn off sourcemaps:
```bash
ng test --sourcemap=false
```
</code></pre>

File: .gitignore [hunk 18]
```diff
@@ -45,4 +45,6 @@ Thumbs.db
 src/api.config.ts
 version.js
 yarn-error.log
-.vscode
\ No newline at end of file
+.vscode
+/temp/*
```
Comment (olastor) [inline .gitignore]: Were `/temp/` and `*.bat` only for your development? They don't seem to be necessary for the master branch.

File: src/app/meditation/testing/fake-meditation.service.ts [hunk 19]
```diff
@@ -0,0 +1,158 @@
...
+      {
+        '_id': '151',
+        'updatedAt': '2017-06-10T06:20:13.856Z',
+        'sitting': 40,
+        'walking': 0,
+        'createdAt': '2017-06-10T06:16:33.905Z',
+        'end': '2017-06-10T06:56:33.905Z',
+        'user': {
+          '_id': '5',
+          'gravatarHash': '4de3df4fa010cd20c1c226fda92b3b83',
```
Comment (olastor) [inline src/app/meditation/testing/fake-meditation.service.ts]: Those records contain user data from the website. We are not allowed to expose these in any way online. Please anonymize all names and change the gravatar hashes to `d8e8fca2dc0f896fd7cb4cb0031ba249` (or any other arbitrary md5 hash).

File: src/app/question/list-entry/question-list-entry.component.spec.ts [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+  question: any;
+  isAdmin: boolean;
+  mode: number;
+
+  public ngOnInit() {
+    this.isAdmin = false;
+    this.mode = 0;
+
+    this.question = {
+      '_id': '5951b4b7800d4a3a760c1426',
```
Comment (olastor) [inline src/app/question/list-entry/question-list-entry.component.spec.ts]: If this is an id from the website, please change. (also for username, gravatarHash)

File: src/app/user/testing/fake-user.service.ts [hunk 21]
```diff
@@ -0,0 +1,197 @@
...
+
+  /**
+   * Gets the complete profile of the given id or, if null, the
+   * currently logged in user.
+   * @return {Observable<Response>}
+   */
+  public getProfile(id: string = null): Observable<Response> {
+    return TestHelper.fakeResponse({
+      '_id': '5',
+      'gravatarHash': '07eda263b3707f9784714c041ae46be6',
```
Comment (olastor) [inline src/app/user/testing/fake-user.service.ts]: same as above

File: README.md (hunk 17 above)
Comment (jiixing) [inline README.md]: I made this change, thanks

File: .gitignore (hunk 18 above)
Comment (jiixing) [inline .gitignore]: yes,  reverted the file, I guess i shouldn't check it in then only keep it checked out in my folder?

**Comment by olastor** [comment]
@jiixing I moved your fork to a new branch in order to make the travis test work (#502). Please take a look if it is the same. You can also check the build log to see if it all works as it should:

https://travis-ci.org/Sirimangalo/meditation-plus-angular/builds/252549268

**Pull request closed without merging**